Tasks/Common/RemoteDeployer/JobHelper.ps1

```diff
@@ -0,0 +1,189 @@
+$defaultScriptArgumentsValues = @{
+    scriptPath = "";
+    scriptArguments = "";
+    inlineScript = "";
+    inline = $true;
+    workingDirectory = "";
+    errorActionPreference = "continue";
+    ignoreLASTEXITCODE = $false;
+    failOnStdErr = $false;
+    initializationScriptPath = "";
+    sessionVariables = "";
+}
+
+function Run-RemoteScriptJobs {
+    [CmdletBinding()]
+    Param (
+        [System.Management.Automation.Runspaces.PSSession[]] $sessions,
+        [scriptblock] $script,
+        [string] $sessionName,
+        [hashtable] $scriptArgumentsByName,
+        [hashtable[]] $targetMachines,
+        [psobject] $sessionOption,
+        [scriptblock] $outputHandler,
+        [scriptblock] $errorHandler,
+        [string] $logsFolder
+    )
+    Trace-VstsEnteringInvocation -InvocationInfo $MyInvocation -Parameter ""
+    try {
+        $scriptArguments = Get-ScriptArguments -scriptArgumentsByName $scriptArgumentsByName
+        $jobName = [Guid]::NewGuid().ToString()
```

Comment:
$jobName = $env:BUILD_BUILDID 
if($env:RELEASE_RELEASEID){
    $jobName  = $env:RELEASE_RELEASEID 
}
$date = Get-Date -Format o
$jobName = -join($jobName , targetName, "-", $date)

Can you use JobName contain releaseId/buildId, targetName and date as this will be more helpful to debug to issue ?

Reply:
jobName is not used anywhere except when starting the jobs for the target machines. every job has a location which is used to identify the target machine within a session. All jobs are started simultaneously, so jobname cannot have a specific target name in it.